game-core/src/main/java/games/strategy/engine/framework/startup/mc/ClientModel.java

```diff
@@ -379,16 +380,15 @@ public void enablePlayer(final String playerName) {
   }
 
   private void internalPlayerListingChanged(final PlayerListing listing) {
-    SwingUtilities
-        .invokeLater(() -> gameSelectorModel.clearDataButKeepGameInfo(listing.getGameName(), listing.getGameRound(),
-            listing.getGameVersion().toString()));
+    gameSelectorModel.clearDataButKeepGameInfo(listing.getGameName(), listing.getGameRound(),
+            listing.getGameVersion().toString());
     synchronized (this) {
       playersToNodes = listing.getPlayerToNodeListing();
       playersEnabledListing = listing.getPlayersEnabledListing();
       playersAllowedToBeDisabled = listing.getPlayersAllowedToBeDisabled();
       playerNamesAndAlliancesInTurnOrder = listing.getPlayerNamesAndAlliancesInTurnOrderLinkedHashMap();
     }
-    SwingUtilities.invokeLater(() -> listener.playerListChanged());
+    listener.playerListChanged();
```

Comment:
Those events shouldn't be triggered on the EDT.
This reduces complexity by leaving this up to the listener implementation